include/inviwo/core/processors/processorutils.h

```diff
@@ -0,0 +1,105 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#ifndef IVW_PROCESSORUTILS_H
+#define IVW_PROCESSORUTILS_H
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/common/inviwo.h>
+
+namespace inviwo {
+
+class Processor;
+class ProcessorMetaData;
+
+namespace util {
+
+/**
+ * Retrieve the meta data of the processor.
+ */
+IVW_CORE_API const ProcessorMetaData* getMetaData(const Processor* processor);
+
+/**
+ * Retrieve the meta data of the processor.
+ */
+IVW_CORE_API ProcessorMetaData* getMetaData(Processor* processor);
+
+/**
+ * Retrieve the position of the processor.
+ */
+IVW_CORE_API ivec2 getPosition(const Processor* processor);
+
+/**
+ * Set the position of processor to pos
+ */
+IVW_CORE_API void setPosition(Processor* processor, ivec2 pos);
+
+/**
+ * Retrieve the position of the processor.
+ */
+IVW_CORE_API bool isSelected(const Processor* processor);
+
+/**
+ * Set the position of processor to pos
+ */
+IVW_CORE_API void setSelected(Processor* processor, bool selected);
+
+
+struct IVW_CORE_API GridPos {
+    GridPos(int x, int y) : pos_{x, y} {};
+    explicit GridPos(ivec2 pos) : pos_{pos} {}
+
+    operator ivec2() const { return pos_ * ivec2{25, 25}; }
```

Comment:
Thats the grid in the editor...

Reply:
This is a bit of the hack since the gridsize or any such thing is not "known" in core...

Reply:
Ok, perhaps we can declare it as a constexpr so that it gets a description. Also, I think that you could apply the transformation in the constructors.